lib/button.js

```diff
@@ -0,0 +1,326 @@
+              buttonMenuItemContainer.appendChild(content);
+              return;
+            }
+
+            applyButtonMenuItemStyle(buttonMenuItemContainer);
+            buttonMenuItemContainer.innerHTML = content;
+            return;
+          }
+
+          addContent(item.title);
+          item.buttonMenuItemContainer = buttonMenuItemContainer;
+
+          if (item.click) {
+            buttonMenuItemContainer.addEventListener('click', function (ev) {
+              item.click.apply(self, [ev]);
+            });
+          }
+        }
+
+        addItem(item);
+        buttonMenu.appendChild(buttonMenuItemContainer);
+      });
+    }
+
+    function clickIndex(idx) {
+      items[idx].buttonMenuItemContainer.dispatchEvent(new Event('click'));
+    }
+
+    function init() {
+      var loc = {},
+        s = self.scrollOffset(self.canvas);
+
+      if (zIndexTop === undefined) {
+        zIndexTop = self.style.buttonZIndex;
+      }
+
+      createItems();
+      self.createInlineStyle(
+        buttonMenu,
+        'canvas-datagrid-button-menu' + (self.mobile ? '-mobile' : ''),
+      );
+
+      loc.x = pos.left - s.left;
+      loc.y = pos.top - s.top;
+      loc.height = 0;
+      zIndexTop += 1;
+      buttonMenu.style.position = 'absolute';
+      buttonMenu.style.zIndex = zIndexTop;
+      buttonMenu.style.left = loc.x + 'px';
+      buttonMenu.style.top = loc.y + 'px';
+      document.body.appendChild(buttonMenu);
+      rect = buttonMenu.getBoundingClientRect(); // TODO: fix
+
+      if (rect.bottom > window.innerHeight) {
+        loc.y =
+          self.button.wrapper.top -
+          buttonMenu.offsetHeight -
+          self.scrollBox.scrollTop;
+        if (loc.y < 0) {
+          loc.y = self.style.buttonMenuWindowMargin;
+        }
+
+        if (
+          buttonMenu.offsetHeight >
+          window.innerHeight - self.style.buttonMenuWindowMargin
+        ) {
+          buttonMenu.style.height =
+            window.innerHeight - self.style.buttonMenuWindowMargin * 2 + 'px';
+        }
+      }
+
+      if (rect.right > window.innerWidth) {
+        loc.x -=
+          rect.right - window.innerWidth + self.style.buttonMenuWindowMargin;
+      }
+
+      if (loc.x < 0) {
+        loc.x = self.style.buttonMenuWindowMargin;
+      }
+
+      if (loc.y < 0) {
+        loc.y = self.style.buttonMenuWindowMargin;
+      }
+
+      buttonMenu.style.left = loc.x + 'px';
+      buttonMenu.style.top = loc.y + 'px';
+    }
+
+    intf.buttonMenu = buttonMenu;
+    init();
+    intf.clickIndex = clickIndex;
+    intf.rect = rect;
+    intf.items = items;
+
+    intf.dispose = function () {
+      if (buttonMenu.parentNode) {
+        buttonMenu.parentNode.removeChild(buttonMenu);
+      }
+    };
+
+    Object.defineProperty(intf, 'selectedIndex', {
+      get: function get() {
+        return selectedIndex;
+      },
+      set: function set(value) {
+        if (typeof value !== 'number' || isNaN(value || !isFinite(value))) {
```

Comment:
Should `isNaN(value || !isFinite(value))` actually be `isNaN(value) || !isFinite(value)` ?